src/Mvc/Mvc.Core/src/Routing/ControllerLinkGeneratorExtensions.cs

```diff
@@ -36,7 +36,9 @@ public static class ControllerLinkGeneratorExtensions
         /// names from <c>RouteOptions</c>.
         /// </param>
         /// <returns>A URI with an absolute path, or <c>null</c> if a URI cannot be created.</returns>
+#pragma warning disable RS0026 // Do not add multiple public overloads with optional parameters
         public static string GetPathByAction(
+#pragma warning restore RS0026 // Do not add multiple public overloads with optional parameters
```

Comment:
@ajaybhargavb @JamesNK I realize this and src/Mvc/Mvc.Core/src/Routing/PageLinkGeneratorExtensions.cs aren't particularly new. But, can this problem be addressed and not suppressed without breaking back-compat❔ If not, do you suggest we take the hit now❔

Reply:
The only way to resolve the warning is changing method signatures which is a breaking change.

I don't know of any issues caused by the overloads. Suppressing is fine.

@javiercn

Reply:
I would recommend using attribute-style suppressions for methods, with the Justification set to "Required to maintain compatibility".

Reply:
That seems to work but VS tends to crash or ignore me when using attribute suppressions for this issue. I eventually got it to fix one instance and copied my brand new attribute everywhere else. Something may be busted in this area.